lib/ansible/cli/__init__.py

```diff
@@ -525,7 +525,11 @@ def get_host_list(inventory, subset, pattern='all'):
 
         hosts = inventory.list_hosts(pattern)
         if not hosts and no_hosts is False:
-            raise AnsibleError("Specified hosts and/or --limit does not match any hosts")
+            if inventory.is_group_pattern(pattern):
+                # Valid pattern, just empty
+                display.warning("Specified hosts and/or --limit does not match any hosts, only empty group(s)")
```

Comment:
i'm not sure this should not be an error as we still get an empty inventory while parameters for 'non empty' were supplied

Reply:
probably a better error would be 'Specified hosts and/or --limit does not provide any targetable hosts'